playbooks/roles/ufw/tasks/main.yml

```diff
@@ -0,0 +1,54 @@
+---
+- name: Ensure UFW allows SSH
+  ufw: rule="allow"
+    name="OpenSSH"
+
+- name: Ensure UFW is enabled and denies by default
+  ufw: state="enabled"
+    policy="deny"
+    direction="incoming"
+
+- name: Ensure UFW allows L2TP/IPSec
+  ufw: to_port="{{ item }}"
+    proto="udp"
+    rule="allow"
+  with_items:
+    - "500"
+    - "1701"
+    - "4500"
+
+- name: Ensure UFW allows HTTPS
+  ufw: to_port="443"
+    proto="tcp"
+    rule="allow"
+
+- name: Ensure UFW allows OpenVPN DNS from 10.8.0.1
+  ufw: to_port="53"
+    proto="udp"
+    rule="allow"
+    from_ip="10.8.0.1"
+
+- name: Ensure UFW allows OpenVPN
+  ufw: to_port="636"
+    proto="tcp"
+    rule="allow"
+
+- name: Ensure UFW allows Shadowsocks
+  ufw: to_port="8530"
+    proto="tcp"
+    rule="allow"
+
+- name: Ensure UFW allows stunnel
+  ufw: to_port="993"
+    proto="tcp"
+    rule="allow"
+
+- name: Ensure UFW allows TOR bridge
+  ufw: to_port="{{ tor_obfs3_port }}"
+    proto="tcp"
+    rule="allow"
+
+- name: Ensure UFW allows TOR bfs3 pluggable transport
+  ufw: to_port="57366"
```

Comment:
The port numbers for the obfsproxy pluggable transports are randomized and will be different on every server. The tor-bridge role sets a fact called {{ tor_obfs3_port }} that can be used here.

This variable must be used, otherwise this firewall rule will be broken and the obfsproxy pluggable transport will not work.